Mathlib/Order/InitialSeg.lean

```diff
@@ -656,16 +657,30 @@ theorem monotone [PartialOrder α] (f : α ≤i β) : Monotone f :=
 theorem strictMono [PartialOrder α] (f : α ≤i β) : StrictMono f :=
   f.toOrderEmbedding.strictMono
 
-theorem map_isMin [PartialOrder α] (f : α ≤i β) (h : IsMin a) : IsMin (f a) := by
-  intro b hb
+@[simp]
+theorem isMin_apply_iff [PartialOrder α] (f : α ≤i β) : IsMin (f a) ↔ IsMin a := by
+  refine ⟨StrictMono.isMin_of_apply f.strictMono, fun h b hb ↦ ?_⟩
   obtain ⟨x, rfl⟩ := f.mem_range_of_le hb
   rw [f.le_iff_le] at hb ⊢
   exact h hb
 
+alias ⟨_, map_isMin⟩ := isMin_apply_iff
+
 @[simp]
 theorem map_bot [PartialOrder α] [OrderBot α] [OrderBot β] (f : α ≤i β) : f ⊥ = ⊥ :=
   (map_isMin f isMin_bot).eq_bot
 
+@[simp]
+theorem apply_covBy_apply_iff [PartialOrder α] (f : α ≤i β) : f a ⋖ f a' ↔ a ⋖ a' := by
+  refine ⟨CovBy.of_image f.toOrderEmbedding, fun h ↦ ⟨(lt_iff_lt f).2 h.lt, fun b ha hb ↦ ?_⟩⟩
+  obtain ⟨c, rfl⟩ := f.mem_range_of_rel hb
+  rw [lt_iff_lt] at ha hb
+  exact h.2 ha hb
```

Comment:
```suggestion
theorem apply_covBy_apply_iff [PartialOrder α] (f : α ≤i β) : f a ⋖ f a' ↔ a ⋖ a' :=
  Set.OrdConnected.apply_covBy_apply_iff sorry
```

Reply:
There's a one line proof using this and [`IsLowerSet.ordConnected`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/UpperLower/Basic.html#IsLowerSet.ordConnected), but this increases the imports even further.

Reply:
What should we done? Do we just eat the import increase? Do we make a separate file for the handful of upper set/covering lemmas on initial/principal segments? Do we keep the older and slightly longer proof?

Reply:
The best would be to rearrange the files so that there's little to no import increase, and the second best would be to create a new file

Reply:
I don't really know how to do the former. I think I'll just make a `SuccPred/InitialSeg` file to put all of the covering/successor lemmas in.